packages/web-app-files/tests/unit/components/SideBar/Actions/FileActions.spec.js

```diff
@@ -0,0 +1,411 @@
+    selector: '.oc-files-actions-sidebar-move-trigger',
+    handler: jest.fn()
+  },
+  delete: {
+    class: 'oc-files-actions-sidebar-delete-trigger',
+    selector: '.oc-files-actions-sidebar-delete-trigger',
+    handler: jest.fn()
+  }
+}
+
+const extraActions = {
+  'markdown-editor': {
+    class: 'oc-files-actions-sidebar-markdown-editor-trigger',
+    selector: '.oc-files-actions-sidebar-markdown-editor-trigger',
+    handler: jest.fn()
+  },
+  'draw-io': {
+    class: 'oc-files-actions-sidebar-draw-io-trigger',
+    selector: '.oc-files-actions-sidebar-draw-io-trigger',
+    handler: jest.fn()
+  },
+  mediaviewer: {
+    class: 'oc-files-actions-sidebar-mediaviewer-trigger',
+    selector: '.oc-files-actions-sidebar-mediaviewer-trigger',
+    handler: jest.fn()
+  },
+  'open-folder': {
+    class: 'oc-files-actions-sidebar-navigate-trigger',
+    selector: '.oc-files-actions-sidebar-navigate-trigger',
+    handler: jest.fn()
+  }
+}
+
+const getActions = function(extractions = []) {
+  const actions = []
+  for (const key in basicActions) {
+    const action = basicActions[key]
+
+    const actionObj = {
+      icon: key,
+      handler: action.handler,
+      isEnabled: () => true,
+      label: () => key,
+      componentType: 'oc-button',
+      class: action.class
+    }
+    if (key === 'download') {
+      actionObj.canBeDefault = true
+    }
+    actions.push(actionObj)
+  }
+  for (const key of extractions) {
+    const action = extraActions[key]
+    actions.push({
+      icon: key,
+      handler: action.handler,
+      isEnabled: () => true,
+      label: () => key,
+      componentType: 'oc-button',
+      class: action.class,
+      canBeDefault: true
+    })
+  }
+  return actions
+}
+
+describe('FileActions', () => {
+  describe('when user on personal route', () => {
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('should show Actions buttons on md file', async () => {
+      const actions = { ...basicActions, 'markdown-editor': extraActions['markdown-editor'] }
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'welcome',
+        extension: 'md',
+        extraActions: ['markdown-editor']
+      })
+
+      for (const button in actions) {
+        const action = actions[button]
+
+        const buttonElement = wrapper.find(action.selector)
+        expect(buttonElement.exists()).toBeTruthy()
+
+        await buttonElement.trigger('click.stop')
+
+        const args = action.handler.mock.calls[0]
+
+        expect(action.handler).toHaveBeenCalledTimes(1)
+        expect(args.length).toEqual(2)
+        expect(args[0]).toMatchObject({
+          id: '4',
+          fileId: '4',
+          icon: 'file',
+          name: 'welcome.md',
+          extension: 'md',
+          path: '/welcome.md',
+          type: 'file',
+          ownerDisplayName: 'user1',
+          ownerId: 'user1'
+        })
+        expect(args[1]).toEqual(undefined)
```

Comment:
what is this second argument? Why is this undefined? Is this expected?

Reply:
the second arg given is `action.handlerdata` 
https://github.com/owncloud/web/blob/master/packages/web-app-files/src/components/SideBar/Actions/FileActions.vue#L8

but the implementation of the function only takes one argument
https://github.com/owncloud/web/blob/master/packages/web-app-files/src/mixins/actions/download.js#L28

Reply:
can you add a comment here?